### ISSUE TITLE
Using `setIdleTimeout` causes duplicate filter exception

### ISSUE THREAD

**Issue by ctron** [vert-x3/vertx-mqtt#78]
Setting the idle timeout for a client using:

~~~java
final MqttClientOptions options = new MqttClientOptions();
options.setIdleTimeout(10);
~~~

Causes the following exception:

~~~
14:31:09.663 [vert.x-eventloop-thread-1] ERROR io.vertx.core.impl.ContextImpl - Unhandled exception
java.lang.IllegalArgumentException: Duplicate handler name: idle
	at io.netty.channel.DefaultChannelPipeline.checkDuplicateName(DefaultChannelPipeline.java:1071)
	at io.netty.channel.DefaultChannelPipeline.filterName(DefaultChannelPipeline.java:302)
	at io.netty.channel.DefaultChannelPipeline.addBefore(DefaultChannelPipeline.java:259)
	at io.netty.channel.DefaultChannelPipeline.addBefore(DefaultChannelPipeline.java:249)
	at io.vertx.mqtt.impl.MqttClientImpl.initChannel(MqttClientImpl.java:608)
	at io.vertx.mqtt.impl.MqttClientImpl.lambda$doConnect$2(MqttClientImpl.java:180)
	at io.vertx.core.net.impl.NetClientImpl.lambda$connect$2(NetClientImpl.java:118)
	at io.vertx.core.net.impl.NetClientImpl.lambda$null$9(NetClientImpl.java:256)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:342)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:200)
	at io.vertx.core.net.impl.NetClientImpl.lambda$connected$10(NetClientImpl.java:252)
	at io.vertx.core.net.impl.VertxHandler.setConnection(VertxHandler.java:50)
	at io.vertx.core.net.impl.VertxNetHandler.handlerAdded(VertxNetHandler.java:44)
	at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:606)
	at io.netty.channel.DefaultChannelPipeline.addLast(DefaultChannelPipeline.java:235)
	at io.netty.channel.DefaultChannelPipeline.addLast(DefaultChannelPipeline.java:201)
	at io.vertx.core.net.impl.NetClientImpl.connected(NetClientImpl.java:262)
	at io.vertx.core.net.impl.NetClientImpl.lambda$doConnect$7(NetClientImpl.java:217)
	at io.vertx.core.net.impl.ChannelProvider.lambda$connect$0(ChannelProvider.java:41)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:507)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:500)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:479)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:420)
	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:82)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:306)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:341)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:633)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.base/java.lang.Thread.run(Thread.java:844)
~~~

**Comment by Sammers21** [comment]
I see the problem: by default NetClient put IdleStatehandler into pipeline if _idleTimeout_ option is specified:      
 https://github.com/eclipse/vert.x/blob/master/src/main/java/io/vertx/core/net/impl/NetClientImpl.java#L102

I think that _setIdleTimeout_ do the same thing as _setKeepAliveTimeSeconds_ so we can remove _setKeepAliveTimeSeconds_ or override _setIdleTimeout_ in a way to do the same as _setKeepAliveTimeSeconds_ do

**Comment by Sammers21** [comment]
@ppatierno, what do you think?

**Comment by Sammers21** [comment]
I have just submitted a quick fix for it

**Comment by vietj** [comment]
as precaution I think that the code should be more protective and set the idle timeout option to `0` when constructing the channel

**Comment by vietj** [comment]
i.e :

this.client = vertx.createNetClient(options);

should copy the option and set idle timeout to 0 so we are sure not issue happens.

**Comment by Sammers21** [comment]
@vietj, good point, I have just added this changes to #79 PR.

**Comment by ppatierno** [comment]
@Sammers21 @vietj I'm fine with the fix.

**Comment by ppatierno** [comment]
Closed by #79
